timetable/Objects/Constraints/AbstractConstraint.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Xml.Linq;
+using Timetable.timetable.DB;
+using System.Linq;
+
+namespace Timetable.timetable.Objects
+{
+	public abstract class AbstractConstraint
+	{
+		public int weight { get; set; }
+		public XElement constraint { get; set; }
+
+		/// <summary>
+		/// Sets the weight.
+		/// </summary>
+		/// <param name="w">The width.</param>
+		public void SetWeight(int w)
+		{
+			weight = w;
+			constraint.Add(new XElement("Weight_Percentage", weight));
+
+		}
+		/// <summary>
+		/// Sets the element.
+		/// </summary>
+		/// <param name="s">S.</param>
+		public void SetElement(string s)
+		{
+			constraint = new XElement(s);
+		}
+		/// <summary>
+		/// Returns the XElement
+		/// </summary>
+		/// <returns>The xelement.</returns>
+		public abstract XElement ToXelement();
```

Comment:
Can we implement this in the abstract class? Derived classes just call `return constraint;` here. (That actually makes this method a getter for the `constraint` property which we already have by `public XElement constraint { get; set; }`.

Reply:
No, because it does not just return constraint. It makes the XElement and returns that. That is because each constraint has a different form, different elements.